From a7688eff345133147984507c35bf444131cc3d63 Mon Sep 17 00:00:00 2001
From: Petr Mladek <pmladek@suse.cz>
Date: Tue, 16 Jul 2013 14:45:31 +0000
Subject: fix build with system poppler-0.23.0

XRef *xref parameter has been added to the OutputDev::startPage method, see
http://cgit.freedesktop.org/poppler/poppler/commit/?id=8eb489c355d734a72e140ce7e32470d048362499

Change-Id: I7ad875bf58f6998e639c740f0dac3795738cd1db
Reviewed-on: https://gerrit.libreoffice.org/4940
Reviewed-by: Andras Timar <atimar@suse.com>
Tested-by: Andras Timar <atimar@suse.com>
---
diff --git a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
index 03adf89..53a77d3 100644
--- a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
+++ b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
@@ -481,7 +481,11 @@ PDFOutDev::PDFOutDev( PDFDoc* pDoc ) :
 {
 }
 
-void PDFOutDev::startPage(int /*pageNum*/, GfxState* state)
+void PDFOutDev::startPage(int /*pageNum*/, GfxState* state
+#if POPPLER_CHECK_VERSION(0, 23, 0)
+                          , XRef* /*xref*/
+#endif
+)
 {
     assert(state);
     printf("startPage %f %f\n",
diff --git a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx
index c0977cb7..4f3d3a9 100644
--- a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx
+++ b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx
@@ -181,7 +181,11 @@ namespace pdfi
         virtual void setDefaultCTM(double *ctm) SAL_OVERRIDE;
 
         // Start a page.
-        virtual void startPage(int pageNum, GfxState *state) SAL_OVERRIDE;
+        virtual void startPage(int pageNum, GfxState *state
+#if POPPLER_CHECK_VERSION(0, 23, 0)
+                               , XRef *xref
+#endif
+        ) SAL_OVERRIDE;
 
         // End a page.
         virtual void endPage() SAL_OVERRIDE;
--
cgit v0.9.0.2-2-gbebe

